packages/components/src/advertisement/index.jsx

```diff
@@ -0,0 +1,103 @@
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'clsx';
+import Script from 'react-load-script';
+import { getUID } from 'nebenan-helpers/lib/calculations';
+import { invoke } from 'nebenan-helpers/lib/utils';
+import { getOptions } from './utils';
+
+import styles from './index.module.scss';
+
+const AD_SCRIPT_CDN = 'https://cdn.adnuntius.com/adn.js';
+
+const Advertisement = ({
+  className,
+
+  userId,
+  sessionId,
+  env,
+
+  id,
+  width,
+  height,
+  categories,
+
+  options,
+
+  onRequest,
+  onLoad,
+}) => {
+  const [uid, setUID] = useState(null);
+  const targetClass = `adn-${uid}`;
+
+  useEffect(() => {
+    if (!uid || !window.adn || !id) return;
+
+    const props = getOptions({
+      userId,
+      sessionId,
+      env,
+      auW: width,
+      auH: height,
+      c: categories,
+    });
+
+    const requestOptions = {
+      ...options,
+      ...props,
+
+      targetClass,
+      auId: id,
+
+      isolateFrame: true, // improves sandboxing, may cause issues with some ads
+      useCookies: false, // redundant, added just in case
+      protocol: 'https',
+    };
+
+    window.adn.request(requestOptions);
+    invoke(onRequest, uid, requestOptions);
+  }, [uid]); // Intentionally only loads ad once
```

Comment:
Let's add all the used props in here. It should work right out of the box. I always expect react components to react to prop changes

Reply:
won't work here since adserver will load a new ad every time. we need to make it sticky. this just reacts to adscript being loaded basically, the rest is never changing.